pkg/dataconn/wire.go

```diff
@@ -91,3 +102,14 @@ func (w *Wire) Read() (*Message, error) {
 func (w *Wire) Close() error {
 	return w.conn.Close()
 }
+
+func getRequestHeaderSize() int {
+	var msg Message
+
+	return int(unsafe.Sizeof(msg.MagicVersion)) +
+		int(unsafe.Sizeof(msg.Seq)) +
+		int(unsafe.Sizeof(msg.Type)) +
+		int(unsafe.Sizeof(msg.Offset)) +
+		int(unsafe.Sizeof(msg.Size)) +
+		4 // int(len(msg.Data))
```

Comment:
LGTM.

After discussing with @derekbit it's better to explain why we can assume the size of data is 4 bytes here.

Reply:
Updated. Thank you.